### PR TITLE
Add contextual help tooltips across the UI

### LINKED ISSUE #6279: Add contextual help tooltips with docs links throughout the UI
## Summary

Add contextual help (?) icons throughout the UI that show brief explanations on hover and optionally link to relevant documentation pages. This improves discoverability and reduces the need to leave the app to look up concepts.

## Approach

A shared `HelpTooltip` component standardizes the pattern:

```tsx
<HelpTooltip docsPath="gateway/guides/episodes">
  Episodes group related inferences from multi-step workflows.
</HelpTooltip>
```

- Light gray `CircleHelp` icon (`text-muted-foreground`, 14px)
- Tooltip appears on hover with brief explanation
- When `docsPath` is provided, clicking the icon opens the docs page in a new tab
- `SectionHeader` now accepts a `help` prop for easy integration with section headings

### Design principles

- **Selective, not exhaustive** — only where concepts are genuinely non-obvious
- **Brief and actionable** — 1-2 sentences, not documentation dumps
- **Co-located content** — tooltip text lives next to the UI element it explains (no centralized constants file that creates indirection)
- **Consistent pattern** — same component everywhere, same icon size, same interaction

### High priority (most confusing to new users)

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **SFT form** | Validation Split (%) | Percentage of training data reserved for validation to detect overfitting | `optimization/supervised-fine-tuning-sft` |
| **SFT form** | Max. Samples | Maximum training examples to use, limited by curated inferences meeting threshold | `optimization/supervised-fine-tuning-sft` |
| **SFT form** | Metric/threshold (curation) | Minimum metric score for an inference to be included as training data | `optimization/supervised-fine-tuning-sft` |
| **Evaluation modal** | Inference Cache | Controls whether evaluation results are cached/reused | `gateway/guides/inference-caching` |
| **Evaluation modal** | Concurrency | Number of datapoints evaluated in parallel (already existed, now uses shared component) | — |
| **Evaluation modal** | Adaptive Stopping Precision | Stop evaluator when confidence interval is within threshold (already existed) | — |
| **Function detail** | Experimentation section | How traffic is distributed across variants via weights | `experimentation/run-static-ab-tests` |
| **Episodes page** | Page header | What episodes are (multi-step workflow sequences) | `gateway/guides/episodes` |
| **Models page** | Latency section | Explanation of P50/P90/P95 percentile metrics | — |
| **Dataset builder** | Output source selector | Difference between None/Inference/Demonstration outputs | `gateway/api-reference/datasets-datapoints` |

### Medium priority

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **Function detail** | Variant weights in table | Percentage of traffic sent to this variant | `experimentation/run-static-ab-tests` |
| **Function detail** | Throughput section | Number of inference requests per time period | — |
| **Workflow evaluations** | Page header | How workflow evaluations differ from inference evaluations | `evaluations/workflow-evaluations/tutorial` |
| **Inference detail** | Model Inferences section | Why one inference can produce multiple model inferences | — |
| **Dataset builder** | Episode ID filter | Filter to inferences from a specific multi-step sequence | `gateway/guides/episodes` |
| **Playground** | Reference Output | Expected output from the dataset for comparison | — |
| **API keys** | Public ID column | The non-secret identifier for API key usage | `operations/set-up-auth-for-tensorzero` |

### Lower priority (straightforward but nice to have)

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **Models page** | Usage section | Number of inference requests to each model over time | — |
| **Models/Functions** | Time granularity selector | How metrics are aggregated over time | — |
| **Inference detail** | Demonstration feedback | Human-provided reference outputs used to guide model behavior | `gateway/guides/metrics-feedback` |
| **Function detail** | Schemas section | Input/output schemas defining the function API contract | `gateway/configure-functions-and-variants` |
| **Datasets** | Page header | What datasets are and how they're used for evaluation/optimization | `gateway/api-reference/datasets-datapoints` |

## What NOT to add help tooltips to

- **Self-explanatory labels** — "Function", "Variant", "Time", "ID" don't need explanation
- **Standard UI patterns** — search bars, pagination, sort controls
- **Every column header** — only where the metric/concept is ambiguous
- **Sidebar navigation** — page names are clear enough

## Implementation notes

- The `HelpTooltip` component is already built and the initial batch is implemented in the companion PR
- New tooltips can be added incrementally per-page without coordination
- No centralized constants file — each tooltip text is co-located with its usage, keeping code greppable and avoiding a layer of indirection
- Docs base URL (`https://www.tensorzero.com/docs`) is defined once in the component

### PR DESCRIPTION
**Pull request by simeonlee**
## Summary

Adds contextual help tooltips (using the `HelpTooltip` component from #6282) to places where concepts may not be immediately obvious.

## New tooltips

| Location | Text | Docs link |
|----------|------|-----------|
| Function: Experimentation | How traffic is distributed across variants via weights | A/B testing guide |
| Dataset Builder: Output Source | "None"/"Inference"/"Demonstration" distinction | Datasets API reference |
| Eval: Inference Cache | Reuse vs fresh inferences | Inference caching guide |
| SFT: Validation Split | Data reserved for detecting overfitting | SFT docs |
| SFT: Max Samples | Limited by curated inferences meeting threshold | SFT docs |

## PR stack

1. #6282 — Component + SectionHeader `help` prop + Episodes example
2. #6284 — Refactor existing evaluation tooltips to use HelpTooltip
3. **This PR** — Add new tooltips across the UI

## Open questions for reviewers

- Are the SFT tooltips (Validation Split, Max Samples) useful, or obvious enough to ML practitioners?
- Is the Inference Cache tooltip adding value, or are On/Off/Read Only/Write Only self-explanatory?

## Test plan

- [ ] Hover over (?) icons to verify hover card appears and stays open
- [ ] Click "View docs" links to verify they open correct docs pages in new tab

Closes #6279

---

> [!NOTE]
> **Low Risk**
> UI-only additions/refactors to tooltip/help copy and docs links; low risk aside from minor layout/UX regressions if the tooltip component behaves unexpectedly in these contexts.
> 
> **Overview**
> Adds contextual `HelpTooltip` hovercards (often with docs links via `docsUrl`) to several UI surfaces to clarify concepts like dataset output source selection, evaluation parameters (concurrency, max datapoints, inference cache, adaptive stopping precision), experimentation variant weights, and SFT advanced parameters.
> 
> Refactors existing evaluation help icons/tooltips to use the shared `HelpTooltip` component, removing inline `TooltipProvider`/`TooltipContent` wiring and standardizing help affordances across these screens.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0fc71e966b59656f19ebc1896bb0bfbd71201200. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by simeonlee** [comment]
@GabrielBianconi Fixed a merge conflict, would love a stamp when you get a moment!
